spring-session/src/main/java/org/springframework/session/MapSession.java

```diff
@@ -150,11 +155,17 @@ public void setAttribute(String attributeName, Object attributeValue) {
 		}
 		else {
 			this.sessionAttrs.put(attributeName, attributeValue);
+			this.changed = true;
 		}
 	}
 
 	public void removeAttribute(String attributeName) {
 		this.sessionAttrs.remove(attributeName);
+		this.changed = true;
```

Comment:
I wonder if we should mark this as `changed` if `this.sessionAttrs.remove(attributeName)` returns `null`?